src/ip/QosConfig.cc

```diff
@@ -533,7 +533,11 @@ Ip::Qos::setSockTos(const int fd, tos_t tos, int type)
     // unless we pass it a 32-bit variable storing 8-bits of data.
     // NP: it is documented as 'int' for all systems, even those like Linux which accept 8-bit char
     //     so we convert to a int before setting.
```

Comment:
This comment will become stale if this PR changes are committed as is. If the above claim is still valid for all non-Windows environments (that we know about), then we can simply move this comment inside the new `#else` section and replace "all systems" with "all non-Windows systems" or even "all other systems".

When adjusting the comment, please drop the NP prefix and the related indentation.

```
       IP_TOS (since Linux 1.0)
              Set or receive the Type-Of-Service (TOS) field that is
              sent with every IP packet originating from this socket.
              It is used to prioritize packets on the network.  TOS is a
              byte.  There are some standard TOS flags defined:
```

FWIW, at least some modern Linux [manpages](https://www.man7.org/linux/man-pages/man7/ip.7.html) document the TOS value as a "byte" not "8-bit char". Internally, Linux kernel [probably](https://gist.github.com/wenjianhn/0f7a9a1e36018a42515c?permalink_comment_id=4302154#gistcomment-4302154) stores these values using (the equivalent of) uint8_t. `tos_t` is (the equivalent of) uint8_t. None of that matches "8-bit char" exactly IMHO.

I would also replace "it" with "the underlying socket option value" or something like that to emphasize that we are talking about the option-specific thing the optvalue pointer points at rather than the (generic) optvalue pointer. The latter is `const void *` on modern Linuxes (at least).